feat: opt metrics-server into ambient

noahpb opened the pull request:
## Description
Opts metrics-server into istio ambient mode via the Package CR. Also adds a new e2e test for metrics-server to validate that node and pod metrics are being collected.

## Related Issue
Fixes #1281

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Steps to Validate
```
$ uds run test-single-layer --set FLAVOR=upstream --set LAYER=metrics-server
```

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md) followed

noahpb committed:
feat: add e2e test for metrics-server

noahpb committed:
fix copy/paste mistake

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

noahpb committed:
cleanup e2e test

noahpb committed:
opt in to ambient, add e2e test task

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

slaskawi committed:
feat: task cleanup for Keycloak (#1448)

## Description

This Pull Request cleans up tasks related to Keycloak as they were moved
to UDS Common repo.

This Pull Request depends on
https://github.com/defenseunicorns/uds-common/pull/462. I will mark it
as ready for review once the UDS Common PR gets merged and I can switch
to `main` branch here.

## Related Issue

Relates to:
https://github.com/defenseunicorns/uds-identity-config/issues/373

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Steps to Validate

N/A

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor
Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)
followed

---------

Co-authored-by: UnicornChance <chance@defenseunicorns.com>
Co-authored-by: Chance <139784371+UnicornChance@users.noreply.github.com>

github-actions[bot] committed:
chore(main): release 0.40.0 (#1407)

:robot: I have created a release *beep* *boop*
---


##
[0.40.0](https://github.com/defenseunicorns/uds-core/compare/v0.39.0...v0.40.0)
(2025-04-14)


### ⚠ BREAKING CHANGES

* ServiceMonitors and PodMonitors no longer require TLS configuration
for Istio, and may fail to scrape metrics if TLS configuration is
present. The UDS Operator will handle removing this configuration from
monitors in most cases, but may not update your monitor if TLS
configuration was directly added separate from the Operator's mutations.
In addition, the `istio-certs` and `exempt` scrape classes are no longer
supplied as part of the Prometheus setup and should not be set on your
monitoring resources going forward.
* `Package` CR validation will now prevent creating multiple `Package`
CRs in the same namespace. Ensure that you only have a single `Package`
CR per namespace before this upgrade, otherwise you may be unable to
update them going forward.
* Istio Ambient workloads are now included by default with UDS Core.
These workloads are now part of the `istio-controlplane` component
(previously part of the optional `istio-ambient` component) - any
override values/configuration should target this component instead of
`istio-ambient`.
* Theming configuration for removing additional registration fields has
moved under the `themeCustomizations` values
(`settings.enableRegistrationFields`). If overriding
`DISABLE_REGISTRATION_FIELDS` under `realmInitEnv`, you will need to
switch to this new value.

### Features

* add serviceMesh.mode in Package CR
([#1386](https://github.com/defenseunicorns/uds-core/issues/1386))
([7e50b5d](https://github.com/defenseunicorns/uds-core/commit/7e50b5d591a93e1c6117f52d45f48b37950ef3f1))
* escape slashes in Keycloak Group names
([#1433](https://github.com/defenseunicorns/uds-core/issues/1433))
([6b6be2d](https://github.com/defenseunicorns/uds-core/commit/6b6be2d66f64b80b3ac3e5389c1a5f7c1c5b2e99))
* make istio ambient components default in uds core
([#1428](https://github.com/defenseunicorns/uds-core/issues/1428))
([32d2752](https://github.com/defenseunicorns/uds-core/commit/32d275223c4e4ad305ab6a8e3f902133743fc1c0))
* only allow creation of one `UDSPackage` per namespace
([#1372](https://github.com/defenseunicorns/uds-core/issues/1372))
([2f4dbac](https://github.com/defenseunicorns/uds-core/commit/2f4dbac60ee17afe07a4ce62f96285a0e0f04e47))
* opt prometheus stack into ambient
([#1445](https://github.com/defenseunicorns/uds-core/issues/1445))
([793ccb8](https://github.com/defenseunicorns/uds-core/commit/793ccb87642bad0062cdd92de65f319829aa7001))
* recovering lost Keycloak credentials
([#1410](https://github.com/defenseunicorns/uds-core/issues/1410))
([0f3b536](https://github.com/defenseunicorns/uds-core/commit/0f3b536015ae28e153956bdd26989be33257b7e5))
* task cleanup for Keycloak
([#1448](https://github.com/defenseunicorns/uds-core/issues/1448))
([5af6f2b](https://github.com/defenseunicorns/uds-core/commit/5af6f2b0bf0eecda021800693e4c8ba11e0c1d45))


### Bug Fixes

* authpol remoteserviceaccount enablement
([#1415](https://github.com/defenseunicorns/uds-core/issues/1415))
([c6ae565](https://github.com/defenseunicorns/uds-core/commit/c6ae565bbaf2b958ccf01a1e292ee9038bef7f5f))
* conditional pepr build in tasks
([#1414](https://github.com/defenseunicorns/uds-core/issues/1414))
([ea75df2](https://github.com/defenseunicorns/uds-core/commit/ea75df2ad409ff48f9859aeb848c8af2df2434d6))
* make exemptions conditional for `dev-setup`
([#1442](https://github.com/defenseunicorns/uds-core/issues/1442))
([4d7b471](https://github.com/defenseunicorns/uds-core/commit/4d7b471c1c15622871ecd537710c6096317f148e))
* move disable registration fields to theme values
([#1397](https://github.com/defenseunicorns/uds-core/issues/1397))
([61c67f0](https://github.com/defenseunicorns/uds-core/commit/61c67f0731734b92e96842163f48a7319d3b2156))
* remove flavor from dev deploy of prom CRDs task
([#1419](https://github.com/defenseunicorns/uds-core/issues/1419))
([10c9ff2](https://github.com/defenseunicorns/uds-core/commit/10c9ff23b74c8f22e47e53767ea963368d7f381d))


### Miscellaneous

* **ci:** add e2e tests for cloud distros
([#1259](https://github.com/defenseunicorns/uds-core/issues/1259))
([b116a96](https://github.com/defenseunicorns/uds-core/commit/b116a96359406f80d5ea4347d78bd77c14c4ea5a))
* **deps:** update istio to v1.25.1
([#1387](https://github.com/defenseunicorns/uds-core/issues/1387))
([c538ef4](https://github.com/defenseunicorns/uds-core/commit/c538ef4a5671f8157880422cd88baa8e304f4f9b))
* **deps:** update loki
([#1349](https://github.com/defenseunicorns/uds-core/issues/1349))
([f087f55](https://github.com/defenseunicorns/uds-core/commit/f087f55b64cd13578e9b9d1c652ae69ca1a0c694))
* **deps:** update loki to v3.4.3
([#1426](https://github.com/defenseunicorns/uds-core/issues/1426))
([cc7fbd1](https://github.com/defenseunicorns/uds-core/commit/cc7fbd198fa429e8dd48fafa61bca8cf8742b480))
* **deps:** update neuvector to 5.4.3
([#1368](https://github.com/defenseunicorns/uds-core/issues/1368))
([6c4b44e](https://github.com/defenseunicorns/uds-core/commit/6c4b44e969aa633726f0ed70f2e5ccf7c88dcc9b))
* **deps:** update prometheus-stack
([#1402](https://github.com/defenseunicorns/uds-core/issues/1402))
([707b07d](https://github.com/defenseunicorns/uds-core/commit/707b07d4f17a872859c77cf9480433f277a006fa))
* **deps:** update support dependencies to v3.28.14
([#1435](https://github.com/defenseunicorns/uds-core/issues/1435))
([d29d1b5](https://github.com/defenseunicorns/uds-core/commit/d29d1b5abfe484747631b53b7475af83fa7946d0))
* **deps:** update support dependencies to v3.28.15
([#1441](https://github.com/defenseunicorns/uds-core/issues/1441))
([1e7ebce](https://github.com/defenseunicorns/uds-core/commit/1e7ebce255256c3b3bb7a5407acdd43473fb918f))
* **deps:** update support dependencies to v3.4.8
([#1450](https://github.com/defenseunicorns/uds-core/issues/1450))
([598242b](https://github.com/defenseunicorns/uds-core/commit/598242bdcc77f164d124459a169729905d933394))
* **deps:** update support dependencies to v4.6.1
([#1451](https://github.com/defenseunicorns/uds-core/issues/1451))
([efb22ab](https://github.com/defenseunicorns/uds-core/commit/efb22ab920b1972a0bfe0d3deb220ad80c2a040c))
* **deps:** update support-deps
([#1409](https://github.com/defenseunicorns/uds-core/issues/1409))
([d1ade16](https://github.com/defenseunicorns/uds-core/commit/d1ade16d282b78f1dd35b9ced2a4979672f534a1))
* **deps:** update support-deps
([#1418](https://github.com/defenseunicorns/uds-core/issues/1418))
([0eecf5f](https://github.com/defenseunicorns/uds-core/commit/0eecf5ff47d0b1388f99261b24f0e919db9b7829))
* **deps:** update support-deps
([#1425](https://github.com/defenseunicorns/uds-core/issues/1425))
([9b6f681](https://github.com/defenseunicorns/uds-core/commit/9b6f6812042aa5345efc69c7242922c0014ae867))
* **deps:** update support-deps
([#1443](https://github.com/defenseunicorns/uds-core/issues/1443))
([05def89](https://github.com/defenseunicorns/uds-core/commit/05def890049a4cb33e19634f93f08bed008cb09f))
* **deps:** update support-deps
([#1455](https://github.com/defenseunicorns/uds-core/issues/1455))
([ccd72cf](https://github.com/defenseunicorns/uds-core/commit/ccd72cf1191a507887fe593d1c962f9af4fd0acd))
* **deps:** update vector
([#1444](https://github.com/defenseunicorns/uds-core/issues/1444))
([d36014d](https://github.com/defenseunicorns/uds-core/commit/d36014ddbfcb90844e0bf984b3fa5ab8b0e33c96))
* **deps:** update velero to v8.7.1
([#1391](https://github.com/defenseunicorns/uds-core/issues/1391))
([ea4ed0f](https://github.com/defenseunicorns/uds-core/commit/ea4ed0fe7aa2282ed6aeec397432f14790461cb9))
* **docs:** fix order of authpols doc
([#1408](https://github.com/defenseunicorns/uds-core/issues/1408))
([ee55ab1](https://github.com/defenseunicorns/uds-core/commit/ee55ab10091f1ef150bc1050b19ca8ec494deaa8))
* prefer `===` for comparisons
([#1412](https://github.com/defenseunicorns/uds-core/issues/1412))
([6963633](https://github.com/defenseunicorns/uds-core/commit/696363385bfc569c79688364ad16d03aae852117))
* reduce sidecar cpu/memory requests for CI single-layer testing
([#1459](https://github.com/defenseunicorns/uds-core/issues/1459))
([cc8c405](https://github.com/defenseunicorns/uds-core/commit/cc8c405af6ce2db0d98b3e9bded66904e9e550fd))
* remove watch and conditional logic around ambient component
([#1447](https://github.com/defenseunicorns/uds-core/issues/1447))
([d519af3](https://github.com/defenseunicorns/uds-core/commit/d519af3d9f90980dfcaa90eed0ecf98cec089111))
* update changelog
([#1406](https://github.com/defenseunicorns/uds-core/issues/1406))
([4239d95](https://github.com/defenseunicorns/uds-core/commit/4239d958744408d47c93021d4d5937b426d36db9))


### Documentation

* fix Velero doc link path
([#1456](https://github.com/defenseunicorns/uds-core/issues/1456))
([01cea57](https://github.com/defenseunicorns/uds-core/commit/01cea5729d3e600e23f04ef131939e3f87983090))

---
This PR was generated with [Release
Please](https://github.com/googleapis/release-please). See
[documentation](https://github.com/googleapis/release-please#release-please).

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

slaskawi commented on src/metrics-server/chart/templates/peerauthentication/metrics-api.yaml:
```diff
@@ -1,20 +0,0 @@
-# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
-
-{{- if .Capabilities.APIVersions.Has "security.istio.io/v1beta1" }}
-apiVersion: security.istio.io/v1beta1
```
Could you please tell me if removing the `PeerAuthentication` has any impact on the security posture? If it lowers it, do we have a remediation plan?

noahpb replied:
Good question. It's my understanding that this `PeerAuthentication` is no longer required as previously it allowed `PERMISSIVE` ingress over port `10250` because calls to metrics server originate from the nodes themselves (and thus, outside of the mesh). In my testing, I found that ambient allowed cluster components to communicate with metrics-server without this workaround. 

Strict mTLS is still [enforced by default](https://github.com/defenseunicorns/uds-core/blob/3423176db0dbbcc17ac9be27a135f7d16939be5a/src/istio/common/chart/templates/peer-authentication.yaml#L13) across the entire mesh via the default `PeerAuthentication` that we create in the `istio-system` namespace, so there is no risk with removing this.

slaskawi replied:
Perfect! Thanks for explaining!

noahpb replied:
I've reverted this change as the `PeerAuthentication` is required for multi node clusters.

chance-coleman commented on test/jest/metrics-server.spec.ts:
```diff
@@ -0,0 +1,121 @@
+    const https = require('https');
+
+    if (!apiServerAddress) {
+        console.error("API server address not found.");
```
I might be misunderstanding this, if the apiServerAddress or the certs are missing we should fail the tests right? if so i think it would make more sense to throw an error. It would "fail fast", preventing jest from silently skipping test logic and being a bit more loud in CI if there is a failure.

```suggestion
        if (!apiServerAddress) {
            throw new Error("API server address not found.");
        }
```

noahpb replied:
Good callout! Updated to `throw` errors instead of logging them.

chance-coleman commented on test/jest/metrics-server.spec.ts:
```diff
@@ -0,0 +1,121 @@
+    try {
+      const currentContext = kc.getCurrentContext();
+      const context = kc.getContextObject(currentContext);
+  
```
There is a bunch of extra whitespace in this file.

chance-coleman replied:
i realize now this comment is kinda hard to see unless you have plugin or something highlights that extra white space, sorry 😬. All of the extra spaces are in the `getCertsFromKubeConfig` and the `getApiServerAddress` functions which i think you'd be able to see if you highlighted them in your IDE, i use the `Trailing Spaces` vscode plugin for this.

noahpb replied:
TIL, thanks for that tip!

noahpb committed:
address feedback

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

noahpb commented:
~~Moving to draft - this is causing issues on non-k3d clusters.~~

Fixed. We needed the `PeerAuthentication` for multi node clusters (this is not the default setup for our CI or local testing).

slaskawi reviewed (dismissed):
LGTM assuming that testing issues are resolved

renovate[bot] committed:
chore(deps): update grafana to v8.11.1 (#1405)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
| [grafana](https://grafana.com)
([source](https://redirect.github.com/grafana/helm-charts)) | patch |
`8.11.0` -> `8.11.1` |

---

### Release Notes

<details>
<summary>grafana/helm-charts (grafana)</summary>

###
[`v8.11.1`](https://redirect.github.com/grafana/helm-charts/releases/tag/grafana-8.11.1)

[Compare
Source](https://redirect.github.com/grafana/helm-charts/compare/grafana-8.11.0...grafana-8.11.1)

The leading tool for querying and visualizing time series and metrics.

#### What's Changed

- \[grafana] Process sidecar configmap/secret label and labelValue with
tpl by [@&#8203;a-abella](https://redirect.github.com/a-abella) in
[https://github.com/grafana/helm-charts/pull/3585](https://redirect.github.com/grafana/helm-charts/pull/3585)

#### New Contributors

- [@&#8203;a-abella](https://redirect.github.com/a-abella) made their
first contribution in
[https://github.com/grafana/helm-charts/pull/3585](https://redirect.github.com/grafana/helm-charts/pull/3585)

**Full Changelog**:
https://github.com/grafana/helm-charts/compare/grafana-sampling-1.1.5...grafana-8.11.1

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOS4yMjcuMyIsInVwZGF0ZWRJblZlciI6IjM5LjIyNy4zIiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6W119-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

andrewg-xyz committed:
chore(docs): zarf package annotations (#1429)

## Description

Add annotations to Zarf Packages (layers)

---------

Co-authored-by: Chance <139784371+UnicornChance@users.noreply.github.com>
Co-authored-by: Micah Nagel <micah.nagel@defenseunicorns.com>

noahpb committed:
add peerauthentication

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

noahpb committed:
fix lint

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

mjnagel commented on test/jest/metrics-server.spec.ts:
```diff
@@ -0,0 +1,107 @@
+    }
+  }
+
+function getCertsFromKubeConfig(): KubeConfigCerts | undefined {
```
Only concern with this is validating/testing that it works on EKS/RKE2 since we run these tests there as well.

noahpb replied:
Good callout. We can switch to using `client-node` for auth and use that for fetching pod metrics.

noahpb replied:
Refactored to use `client-node`. Tested locally and works against EKS.

renovate[bot] committed:
chore(deps): update neuvector (#1417)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[cgr.dev/du-uds-defenseunicorns/neuvector-controller-fips](https://images.chainguard.dev/directory/image/neuvector-controller-fips/overview)
([source](https://redirect.github.com/chainguard-images/images-private/tree/HEAD/images/neuvector-fips))
| patch | `5.4.2` -> `5.4.3` |
|
[cgr.dev/du-uds-defenseunicorns/neuvector-enforcer-fips](https://images.chainguard.dev/directory/image/neuvector-enforcer-fips/overview)
([source](https://redirect.github.com/chainguard-images/images-private/tree/HEAD/images/neuvector-fips))
| patch | `5.4.2` -> `5.4.3` |
|
[cgr.dev/du-uds-defenseunicorns/neuvector-manager](https://images.chainguard.dev/directory/image/neuvector-manager/overview)
([source](https://redirect.github.com/chainguard-images/images-private/tree/HEAD/images/neuvector))
| patch | `5.4.2` -> `5.4.3` |
| [core](https://neuvector.com)
([source](https://redirect.github.com/neuvector/neuvector-helm)) | patch
| `2.8.4` -> `2.8.5` |
| [crd](https://neuvector.com) | patch | `2.8.4` -> `2.8.5` |
|
[docker.io/neuvector/controller](https://www.suse.com/products/base-container-images/)
([source](https://sources.suse.com/SUSE:SLE-15-SP6:Update:CR/micro-image/19856e79d950c4baf0d9cc9c3e07c2f3/))
| patch | `5.4.2` -> `5.4.3` |
|
[docker.io/neuvector/enforcer](https://www.suse.com/products/base-container-images/)
([source](https://sources.suse.com/SUSE:SLE-15-SP6:Update:CR/micro-image/19856e79d950c4baf0d9cc9c3e07c2f3/))
| patch | `5.4.2` -> `5.4.3` |
|
[docker.io/neuvector/manager](https://www.suse.com/products/base-container-images/)
([source](https://sources.suse.com/SUSE:SLE-15-SP6:Update:CR/micro-image/19856e79d950c4baf0d9cc9c3e07c2f3/))
| patch | `5.4.2` -> `5.4.3` |
| [monitor](https://neuvector.com) | patch | `2.8.4` -> `2.8.5` |
|
[registry1.dso.mil/ironbank/neuvector/neuvector/controller](https://open-docs.neuvector.com/)
([source](https://repo1.dso.mil/dsop/neuvector/neuvector/controller)) |
patch | `5.4.2` -> `5.4.3` |
|
[registry1.dso.mil/ironbank/neuvector/neuvector/enforcer](https://open-docs.neuvector.com/)
([source](https://repo1.dso.mil/dsop/neuvector/neuvector/enforcer)) |
patch | `5.4.2` -> `5.4.3` |
|
[registry1.dso.mil/ironbank/neuvector/neuvector/manager](https://open-docs.neuvector.com/)
([source](https://repo1.dso.mil/dsop/neuvector/neuvector/manager)) |
patch | `5.4.2` -> `5.4.3` |

---

### Release Notes

<details>
<summary>neuvector/neuvector-helm (core)</summary>

###
[`v2.8.5`](https://redirect.github.com/neuvector/neuvector-helm/releases/tag/v2.8.5):
Release 2.8.5

[Compare
Source](https://redirect.github.com/neuvector/neuvector-helm/compare/v2.8.4...v2.8.5)

##### What's Changed

- chore: bump compliance-config version by
[@&#8203;holyspectral](https://redirect.github.com/holyspectral) in
[https://github.com/neuvector/neuvector-helm/pull/469](https://redirect.github.com/neuvector/neuvector-helm/pull/469)
- NVSHAS-9426 Add hostPath for scanner to helm chart by
[@&#8203;venkateshjayagopal](https://redirect.github.com/venkateshjayagopal)
in
[https://github.com/neuvector/neuvector-helm/pull/470](https://redirect.github.com/neuvector/neuvector-helm/pull/470)
- NVSHAS-9748 \[Helm] NV helm update for supporting name referral for
common groups in CRD by
[@&#8203;venkateshjayagopal](https://redirect.github.com/venkateshjayagopal)
in
[https://github.com/neuvector/neuvector-helm/pull/472](https://redirect.github.com/neuvector/neuvector-helm/pull/472)
- chore: update codeowners by
[@&#8203;holyspectral](https://redirect.github.com/holyspectral) in
[https://github.com/neuvector/neuvector-helm/pull/476](https://redirect.github.com/neuvector/neuvector-helm/pull/476)
- chore: update image tags for 5.4.3 by
[@&#8203;holyspectral](https://redirect.github.com/holyspectral) in
[https://github.com/neuvector/neuvector-helm/pull/475](https://redirect.github.com/neuvector/neuvector-helm/pull/475)

**Full Changelog**:
https://github.com/neuvector/neuvector-helm/compare/v2.8.4...v2.8.5

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config
help](https://redirect.github.com/renovatebot/renovate/discussions) if
that's undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOS4yMjcuMyIsInVwZGF0ZWRJblZlciI6IjM5LjIyNy4zIiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6W119-->

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Chance <139784371+UnicornChance@users.noreply.github.com>
Co-authored-by: Micah Nagel <micah.nagel@defenseunicorns.com>
Co-authored-by: Sebastian Łaskawiec <sebastian.laskawiec@defenseunicorns.com>
Co-authored-by: Noah <40781376+noahpb@users.noreply.github.com>
Co-authored-by: Brandi McCall <117415095+bm54cloud@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: UnicornChance <chance@defenseunicorns.com>
Co-authored-by: Noah Birrer <noah@defenseunicorns.com>

renovate[bot] committed:
chore(deps): update support-deps (#1473)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence | Type |
Update |
|---|---|---|---|---|---|---|---|
| [@playwright/test](https://playwright.dev)
([source](https://redirect.github.com/microsoft/playwright)) | [`1.51.1`
->
`1.52.0`](https://renovatebot.com/diffs/npm/@playwright%2ftest/1.51.1/1.52.0)
|
[![age](https://developer.mend.io/api/mc/badges/age/npm/@playwright%2ftest/1.52.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/npm/@playwright%2ftest/1.52.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/npm/@playwright%2ftest/1.51.1/1.52.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/npm/@playwright%2ftest/1.51.1/1.52.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
| devDependencies | minor |
| aws | `~> 5.94.0` -> `~> 5.95.0` |
[![age](https://developer.mend.io/api/mc/badges/age/terraform-provider/hashicorp%2faws/5.95.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/terraform-provider/hashicorp%2faws/5.95.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/terraform-provider/hashicorp%2faws/5.94.1/5.95.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/terraform-provider/hashicorp%2faws/5.94.1/5.95.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
| required_provider | minor |
| ghcr.io/zarf-dev/packages/init | `v0.51.0` -> `v0.52.1` |
[![age](https://developer.mend.io/api/mc/badges/age/docker/ghcr.io%2fzarf-dev%2fpackages%2finit/v0.52.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/docker/ghcr.io%2fzarf-dev%2fpackages%2finit/v0.52.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/docker/ghcr.io%2fzarf-dev%2fpackages%2finit/v0.51.0/v0.52.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/docker/ghcr.io%2fzarf-dev%2fpackages%2finit/v0.51.0/v0.52.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
| | minor |
| mcr.microsoft.com/playwright | `v1.51.1-noble` -> `v1.52.0-noble` |
[![age](https://developer.mend.io/api/mc/badges/age/docker/mcr.microsoft.com%2fplaywright/v1.52.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/docker/mcr.microsoft.com%2fplaywright/v1.52.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/docker/mcr.microsoft.com%2fplaywright/v1.51.1/v1.52.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/docker/mcr.microsoft.com%2fplaywright/v1.51.1/v1.52.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)
| | minor |
| [zarf-dev/zarf](https://redirect.github.com/zarf-dev/zarf) | `v0.51.0`
-> `v0.52.1` |
[![age](https://developer.mend.io/api/mc/badges/age/github-tags/zarf-dev%2fzarf/v0.52.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/github-tags/zarf-dev%2fzarf/v0.52.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/github-tags/zarf-dev%2fzarf/v0.51.0/v0.52.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/github-tags/zarf-dev%2fzarf/v0.51.0/v0.52.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
| | minor |

---

### Release Notes

<details>
<summary>microsoft/playwright (@&#8203;playwright/test)</summary>

###
[`v1.52.0`](https://redirect.github.com/microsoft/playwright/compare/v1.51.1...471930b1ceae03c9e66e0eb80c1364a1a788e7db)

[Compare
Source](https://redirect.github.com/microsoft/playwright/compare/v1.51.1...v1.52.0)

</details>

<details>
<summary>zarf-dev/zarf (zarf-dev/zarf)</summary>

###
[`v0.52.1`](https://redirect.github.com/zarf-dev/zarf/compare/v0.52.0...v0.52.1)

[Compare
Source](https://redirect.github.com/zarf-dev/zarf/compare/v0.52.0...v0.52.1)

###
[`v0.52.0`](https://redirect.github.com/zarf-dev/zarf/releases/tag/v0.52.0):
WIP v0.52.0

[Compare
Source](https://redirect.github.com/zarf-dev/zarf/compare/v0.51.0...v0.52.0)

<!-- Release notes generated using configuration in .github/release.yml
at v0.52.0 -->

##### What's Changed

##### 🚀 Updates

- feat!: Add debug granularity to package create validation and update
error message by [@&#8203;mkcp](https://redirect.github.com/mkcp) in
[https://github.com/zarf-dev/zarf/pull/3636](https://redirect.github.com/zarf-dev/zarf/pull/3636)
- refactor: remove statefulness from Helm package by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3609](https://redirect.github.com/zarf-dev/zarf/pull/3609)
- chore(deps): update to distribution/distribution v3 by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3642](https://redirect.github.com/zarf-dev/zarf/pull/3642)
- fix: oci-concurrency flag on init by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3643](https://redirect.github.com/zarf-dev/zarf/pull/3643)
- docs: Migrate "air gap" to "airgap" by
[@&#8203;mkcp](https://redirect.github.com/mkcp) in
[https://github.com/zarf-dev/zarf/pull/3638](https://redirect.github.com/zarf-dev/zarf/pull/3638)
- chore: update Zarf Homebrew description by
[@&#8203;mkcp](https://redirect.github.com/mkcp) in
[https://github.com/zarf-dev/zarf/pull/3646](https://redirect.github.com/zarf-dev/zarf/pull/3646)
- refactor: include all workload resources by
[@&#8203;wcrum](https://redirect.github.com/wcrum) in
[https://github.com/zarf-dev/zarf/pull/3640](https://redirect.github.com/zarf-dev/zarf/pull/3640)
- chore(deps): update k3s version by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3648](https://redirect.github.com/zarf-dev/zarf/pull/3648)
- refactor: find images by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3621](https://redirect.github.com/zarf-dev/zarf/pull/3621)
- feat: respect filters for all package sources in `LoadPackage` by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3650](https://redirect.github.com/zarf-dev/zarf/pull/3650)
- fix(publish): deterministic SHA for package publish by
[@&#8203;brandtkeller](https://redirect.github.com/brandtkeller) in
[https://github.com/zarf-dev/zarf/pull/3656](https://redirect.github.com/zarf-dev/zarf/pull/3656)
- feat: zarf package inspect manifests by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3653](https://redirect.github.com/zarf-dev/zarf/pull/3653)
- fix: restore ability to publish Zarf packages to root level of an OCI
registry by [@&#8203;Noxsios](https://redirect.github.com/Noxsios) in
[https://github.com/zarf-dev/zarf/pull/3663](https://redirect.github.com/zarf-dev/zarf/pull/3663)
- fix(publish): cleanup trailing manifest file by
[@&#8203;brandtkeller](https://redirect.github.com/brandtkeller) in
[https://github.com/zarf-dev/zarf/pull/3662](https://redirect.github.com/zarf-dev/zarf/pull/3662)
- feat: zarf dev inspect manifest by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3658](https://redirect.github.com/zarf-dev/zarf/pull/3658)
- fix(skeleton): resolve support for kustomizations in isolation by
[@&#8203;brandtkeller](https://redirect.github.com/brandtkeller) in
[https://github.com/zarf-dev/zarf/pull/3669](https://redirect.github.com/zarf-dev/zarf/pull/3669)
- chore: move v1beta1 internal by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3659](https://redirect.github.com/zarf-dev/zarf/pull/3659)
- docs: remove component-choice example by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3668](https://redirect.github.com/zarf-dev/zarf/pull/3668)
- fix: require username and password for `zarf tools registry login` by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3676](https://redirect.github.com/zarf-dev/zarf/pull/3676)
- fix: require one of `password` or `password-stdin` in `zarf tools
registry login` by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3681](https://redirect.github.com/zarf-dev/zarf/pull/3681)
- feat: allow unused flavors with a warning by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3683](https://redirect.github.com/zarf-dev/zarf/pull/3683)
- feature: Add ability for zarf to find oci artifacts for fluxcd ocirepo
resource by [@&#8203;a1994sc](https://redirect.github.com/a1994sc) in
[https://github.com/zarf-dev/zarf/pull/3238](https://redirect.github.com/zarf-dev/zarf/pull/3238)
- feat: warn if cert is expiring soon, error if cert is expired by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/3677](https://redirect.github.com/zarf-dev/zarf/pull/3677)

##### 📦 Dependencies

- chore(deps): bump actions/dependency-review-action from 4.5.0 to 4.6.0
by [@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3626](https://redirect.github.com/zarf-dev/zarf/pull/3626)
- chore(deps): bump github/codeql-action from 3.28.12 to 3.28.15 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3641](https://redirect.github.com/zarf-dev/zarf/pull/3641)
- chore(deps): bump actions/create-github-app-token from 1.12.0 to 2.0.2
by [@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3634](https://redirect.github.com/zarf-dev/zarf/pull/3634)
- chore(deps): bump github.com/defenseunicorns/pkg/oci from 1.0.2 to
1.0.4 by [@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3652](https://redirect.github.com/zarf-dev/zarf/pull/3652)
- chore(deps): bump the cosign-providers group across 1 directory with 3
updates by [@&#8203;dependabot](https://redirect.github.com/dependabot)
in
[https://github.com/zarf-dev/zarf/pull/3651](https://redirect.github.com/zarf-dev/zarf/pull/3651)
- chore(deps): bump the golang group across 1 directory with 3 updates
by [@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3657](https://redirect.github.com/zarf-dev/zarf/pull/3657)
- chore(deps): bump github.com/prometheus/client_golang from 1.21.0 to
1.22.0 by [@&#8203;dependabot](https://redirect.github.com/dependabot)
in
[https://github.com/zarf-dev/zarf/pull/3665](https://redirect.github.com/zarf-dev/zarf/pull/3665)
- chore(deps): bump github.com/golang-cz/devslog from 0.0.11 to 0.0.12
by [@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3664](https://redirect.github.com/zarf-dev/zarf/pull/3664)
- chore(deps): bump actions/setup-node from 4.3.0 to 4.4.0 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3674](https://redirect.github.com/zarf-dev/zarf/pull/3674)
- chore(deps): bump codecov/codecov-action from 5.4.0 to 5.4.2 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3673](https://redirect.github.com/zarf-dev/zarf/pull/3673)
- chore(deps): bump golang.org/x/net from 0.37.0 to 0.38.0 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3688](https://redirect.github.com/zarf-dev/zarf/pull/3688)
- chore(deps): bump helm.sh/helm/v3 from 3.17.2 to 3.17.3 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3689](https://redirect.github.com/zarf-dev/zarf/pull/3689)
- chore(deps): bump github.com/containerd/containerd from 1.7.24 to
1.7.27 by [@&#8203;dependabot](https://redirect.github.com/dependabot)
in
[https://github.com/zarf-dev/zarf/pull/3690](https://redirect.github.com/zarf-dev/zarf/pull/3690)
- chore(deps): bump github.com/golang-jwt/jwt/v4 from 4.5.1 to 4.5.2 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3687](https://redirect.github.com/zarf-dev/zarf/pull/3687)
- chore(deps): bump github.com/go-jose/go-jose/v3 from 3.0.3 to 3.0.4 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/3693](https://redirect.github.com/zarf-dev/zarf/pull/3693)

**Full Changelog**:
https://github.com/zarf-dev/zarf/compare/v0.51.0...v0.52.0

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config
help](https://redirect.github.com/renovatebot/renovate/discussions) if
that's undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOS4yNDguNCIsInVwZGF0ZWRJblZlciI6IjM5LjI0OC40IiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6W119-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Chance <139784371+UnicornChance@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update pepr (#1454)

This PR contains the following updates:

| Package | Type | Update | Change | Age | Adoption | Passing |
Confidence |
|---|---|---|---|---|---|---|---|
|
[defenseunicorns/pepr](https://redirect.github.com/defenseunicorns/pepr)
| | minor | `v0.47.0` -> `v0.48.1` |
[![age](https://developer.mend.io/api/mc/badges/age/github-tags/defenseunicorns%2fpepr/v0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/github-tags/defenseunicorns%2fpepr/v0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/github-tags/defenseunicorns%2fpepr/v0.47.0/v0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/github-tags/defenseunicorns%2fpepr/v0.47.0/v0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
| [lint-staged](https://redirect.github.com/lint-staged/lint-staged) |
devDependencies | patch | [`15.5.0` ->
`15.5.1`](https://renovatebot.com/diffs/npm/lint-staged/15.5.0/15.5.1) |
[![age](https://developer.mend.io/api/mc/badges/age/npm/lint-staged/15.5.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/npm/lint-staged/15.5.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/npm/lint-staged/15.5.0/15.5.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/npm/lint-staged/15.5.0/15.5.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
| [pepr](https://redirect.github.com/defenseunicorns/pepr) |
dependencies | minor | [`0.47.0` ->
`0.48.1`](https://renovatebot.com/diffs/npm/pepr/0.47.0/0.48.1) |
[![age](https://developer.mend.io/api/mc/badges/age/npm/pepr/0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/npm/pepr/0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/npm/pepr/0.47.0/0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/npm/pepr/0.47.0/0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
|
[registry1.dso.mil/ironbank/opensource/defenseunicorns/pepr/controller](https://redirect.github.com/defenseunicorns/pepr)
([source](https://repo1.dso.mil/dsop/opensource/defenseunicorns/pepr/controller))
| | minor | `v0.47.0` -> `v0.48.1` |
[![age](https://developer.mend.io/api/mc/badges/age/docker/registry1.dso.mil%2fironbank%2fopensource%2fdefenseunicorns%2fpepr%2fcontroller/v0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/docker/registry1.dso.mil%2fironbank%2fopensource%2fdefenseunicorns%2fpepr%2fcontroller/v0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/docker/registry1.dso.mil%2fironbank%2fopensource%2fdefenseunicorns%2fpepr%2fcontroller/v0.47.0/v0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/docker/registry1.dso.mil%2fironbank%2fopensource%2fdefenseunicorns%2fpepr%2fcontroller/v0.47.0/v0.48.1?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
| [ts-jest](https://kulshekhar.github.io/ts-jest)
([source](https://redirect.github.com/kulshekhar/ts-jest)) |
devDependencies | patch | [`29.3.1` ->
`29.3.2`](https://renovatebot.com/diffs/npm/ts-jest/29.3.1/29.3.2) |
[![age](https://developer.mend.io/api/mc/badges/age/npm/ts-jest/29.3.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![adoption](https://developer.mend.io/api/mc/badges/adoption/npm/ts-jest/29.3.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![passing](https://developer.mend.io/api/mc/badges/compatibility/npm/ts-jest/29.3.1/29.3.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|
[![confidence](https://developer.mend.io/api/mc/badges/confidence/npm/ts-jest/29.3.1/29.3.2?slim=true)](https://docs.renovatebot.com/merge-confidence/)
|

---

### Release Notes

<details>
<summary>defenseunicorns/pepr (defenseunicorns/pepr)</summary>

###
[`v0.48.1`](https://redirect.github.com/defenseunicorns/pepr/releases/tag/v0.48.1)

[Compare
Source](https://redirect.github.com/defenseunicorns/pepr/compare/v0.48.0...v0.48.1)

##### What's Changed

- fix: resolve formatting error in release script by
[@&#8203;samayer12](https://redirect.github.com/samayer12) in
[https://github.com/defenseunicorns/pepr/pull/2035](https://redirect.github.com/defenseunicorns/pepr/pull/2035)

**Full Changelog**:
https://github.com/defenseunicorns/pepr/compare/v0.48.0...v0.48.1

###
[`v0.48.0`](https://redirect.github.com/defenseunicorns/pepr/releases/tag/v0.48.0)

[Compare
Source](https://redirect.github.com/defenseunicorns/pepr/compare/v0.47.0...v0.48.0)

##### Features

- feat: allow warnings on validation response by
[@&#8203;mjnagel](https://redirect.github.com/mjnagel) in
[https://github.com/defenseunicorns/pepr/pull/1974](https://redirect.github.com/defenseunicorns/pepr/pull/1974)

Approving with Warnings

```ts
When(a.ConfigMap)
  .IsCreatedOrUpdated()
  .InNamespace("pepr-demo")
  .Validate(request => {
    const warnings = [];

    // Check for deprecated fields
    if (request.Raw.data && request.Raw.data["deprecated-field"]) {
      warnings.push("Warning: The 'deprecated-field' is being used and will be removed in future versions");
    }

    // Check for missing app label
    if (!request.HasLabel("app")) {
      warnings.push("Warning: Best practice is to include an 'app' label for resource identification");
    }

    // Return approval with warnings if any were generated
    return request.Approve(warnings.length > 0 ? warnings : undefined);
  });
```

Denying with Warnings

```ts
When(a.ConfigMap)
  .IsCreatedOrUpdated()
  .InNamespace("pepr-demo")
  .Validate(request => {
    // Check for dangerous settings
    if (request.Raw.data && request.Raw.data["dangerous-setting"] === "true") {
      const warnings = [
        "Warning: The 'dangerous-setting' field is set to 'true'",
        "Consider using a safer configuration option"
      ];

      return request.Deny(
        "ConfigMap contains dangerous settings that are not allowed",
        422,
        warnings
      );
    }

    return request.Approve();
  });
```

Warnings will be included in the Kubernetes API response and can be
displayed to users by `kubectl` and other Kubernetes clients, providing
helpful feedback while still enforcing policies.

##### What's Changed :recycle:

- fix: use consistent uuid length limit in docs by
[@&#8203;samayer12](https://redirect.github.com/samayer12) in
[https://github.com/defenseunicorns/pepr/pull/2000](https://redirect.github.com/defenseunicorns/pepr/pull/2000)
- chore: example to validate action by
[@&#8203;cmwylie19](https://redirect.github.com/cmwylie19) in
[https://github.com/defenseunicorns/pepr/pull/2004](https://redirect.github.com/defenseunicorns/pepr/pull/2004)
- chore: account for unique dockerfile by
[@&#8203;cmwylie19](https://redirect.github.com/cmwylie19) in
[https://github.com/defenseunicorns/pepr/pull/1996](https://redirect.github.com/defenseunicorns/pepr/pull/1996)
- chore: build_version_no_exist by
[@&#8203;cmwylie19](https://redirect.github.com/cmwylie19) in
[https://github.com/defenseunicorns/pepr/pull/2006](https://redirect.github.com/defenseunicorns/pepr/pull/2006)
- chore: shellcheck giving incorrect warning on nightlies.sh by
[@&#8203;cmwylie19](https://redirect.github.com/cmwylie19) in
[https://github.com/defenseunicorns/pepr/pull/2009](https://redirect.github.com/defenseunicorns/pepr/pull/2009)
- chore: remove possible premature program exit from watch-processor by
[@&#8203;samayer12](https://redirect.github.com/samayer12) in
[https://github.com/defenseunicorns/pepr/pull/2010](https://redirect.github.com/defenseunicorns/pepr/pull/2010)
- chore: split watch-processor tests to separate test concerns by
[@&#8203;samayer12](https://redirect.github.com/samayer12) in
[https://github.com/defenseunicorns/pepr/pull/2014](https://redirect.github.com/defenseunicorns/pepr/pull/2014)
- chore: finalize example in actions docs by
[@&#8203;cmwylie19](https://redirect.github.com/cmwylie19) in
[https://github.com/defenseunicorns/pepr/pull/2018](https://redirect.github.com/defenseunicorns/pepr/pull/2018)
- chore: reconcile and watch examples in action docs by
[@&#8203;cmwylie19](https://redirect.github.com/cmwylie19) in
[https://github.com/defenseunicorns/pepr/pull/2019](https://redirect.github.com/defenseunicorns/pepr/pull/2019)
- chore: compliance reporting ADR by
[@&#8203;cmwylie19](https://redirect.github.com/cmwylie19) in
[https://github.com/defenseunicorns/pepr/pull/1824](https://redirect.github.com/defenseunicorns/pepr/pull/1824)
- chore: refactor watch-processor tests for clarity by
[@&#8203;samayer12](https://redirect.github.com/samayer12) in
[https://github.com/defenseunicorns/pepr/pull/2013](https://redirect.github.com/defenseunicorns/pepr/pull/2013)
- chore: make uds test use library by
[@&#8203;cmwylie19](https://redirect.github.com/cmwylie19) in
[https://github.com/defenseunicorns/pepr/pull/2024](https://redirect.github.com/defenseunicorns/pepr/pull/2024)
- chore: removes the `migrateAndSetupWatch` out of `store.ts` for
improved testability by
[@&#8203;cmwylie19](https://redirect.github.com/cmwylie19) in
[https://github.com/defenseunicorns/pepr/pull/2027](https://redirect.github.com/defenseunicorns/pepr/pull/2027)

##### Dependabot 🤖

- chore: bump step-security/harden-runner from 2.11.0 to 2.11.1 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2003](https://redirect.github.com/defenseunicorns/pepr/pull/2003)
- chore: bump
[@&#8203;types/node](https://redirect.github.com/types/node) from
22.13.14 to 22.13.17 in the development-dependencies group by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2001](https://redirect.github.com/defenseunicorns/pepr/pull/2001)
- chore: bump actions/dependency-review-action from 4.5.0 to 4.6.0 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2002](https://redirect.github.com/defenseunicorns/pepr/pull/2002)
- chore: bump the development-dependencies group with 2 updates by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2007](https://redirect.github.com/defenseunicorns/pepr/pull/2007)
- chore: bump kubernetes-fluent-client from 3.4.6 to 3.4.7 in the
production-dependencies group by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2012](https://redirect.github.com/defenseunicorns/pepr/pull/2012)
- chore: bump trufflesecurity/trufflehog from 3.88.20 to 3.88.22 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2011](https://redirect.github.com/defenseunicorns/pepr/pull/2011)
- chore: bump library/node from `990d0ab` to `047d633` by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2016](https://redirect.github.com/defenseunicorns/pepr/pull/2016)
- chore: bump trufflesecurity/trufflehog from 3.88.22 to 3.88.23 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2015](https://redirect.github.com/defenseunicorns/pepr/pull/2015)
- chore: bump github/codeql-action from 3.28.13 to 3.28.15 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2021](https://redirect.github.com/defenseunicorns/pepr/pull/2021)
- chore: bump fast-check from 4.0.1 to 4.1.0 in the
development-dependencies group by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2020](https://redirect.github.com/defenseunicorns/pepr/pull/2020)
- chore: bump fast-check from 4.1.0 to 4.1.1 in the
development-dependencies group by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2026](https://redirect.github.com/defenseunicorns/pepr/pull/2026)
- chore: bump library/node from `047d633` to `73da8b4` by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2025](https://redirect.github.com/defenseunicorns/pepr/pull/2025)
- chore: bump library/node from `73da8b4` to `c5bfe90` by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2028](https://redirect.github.com/defenseunicorns/pepr/pull/2028)
- chore: bump undici from 7.7.0 to 7.8.0 in the development-dependencies
group by [@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2031](https://redirect.github.com/defenseunicorns/pepr/pull/2031)
- chore: bump kfc and chainguard images by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2030](https://redirect.github.com/defenseunicorns/pepr/pull/2030)
- chore: bump
[@&#8203;types/node](https://redirect.github.com/types/node) from
22.14.0 to 22.14.1 in the development-dependencies group by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/defenseunicorns/pepr/pull/2032](https://redirect.github.com/defenseunicorns/pepr/pull/2032)

**Full Changelog**:
https://github.com/defenseunicorns/pepr/compare/v0.47.0...v0.48.0

</details>

<details>
<summary>lint-staged/lint-staged (lint-staged)</summary>

###
[`v15.5.1`](https://redirect.github.com/lint-staged/lint-staged/blob/HEAD/CHANGELOG.md#1551)

[Compare
Source](https://redirect.github.com/lint-staged/lint-staged/compare/v15.5.0...v15.5.1)

##### Patch Changes

-
[#&#8203;1533](https://redirect.github.com/lint-staged/lint-staged/pull/1533)
[`5d53534`](https://redirect.github.com/lint-staged/lint-staged/commit/5d535349958a939cd17d87180196bb2b9dfc452b)
Thanks [@&#8203;iiroj](https://redirect.github.com/iiroj)! - Improve
listing of staged files so that *lint-staged* doesn't crash when
encountering an uninitialized submodule. This should result in less
errors like:

        ✖ Failed to get staged files!

</details>

<details>
<summary>kulshekhar/ts-jest (ts-jest)</summary>

###
[`v29.3.2`](https://redirect.github.com/kulshekhar/ts-jest/blob/HEAD/CHANGELOG.md#2932-2025-04-12)

[Compare
Source](https://redirect.github.com/kulshekhar/ts-jest/compare/v29.3.1...v29.3.2)

##### Bug Fixes

- fix: transpile `js` files from `node_modules` whenever Jest asks
([968370e](https://redirect.github.com/kulshekhar/ts-jest/commit/968370e)),
closes
[#&#8203;4637](https://redirect.github.com/kulshekhar/ts-jest/issues/4637)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config
help](https://redirect.github.com/renovatebot/renovate/discussions) if
that's undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOS4yMzguMCIsInVwZGF0ZWRJblZlciI6IjM5LjI0OC40IiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6W119-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Chance <139784371+UnicornChance@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update support dependencies to v4.7.0 (#1477)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
|
[terraform-aws-modules/s3-bucket/aws](https://registry.terraform.io/modules/terraform-aws-modules/s3-bucket/aws)
([source](https://redirect.github.com/terraform-aws-modules/terraform-aws-s3-bucket))
| module | minor | `4.6.1` -> `4.7.0` |

---

### Release Notes

<details>
<summary>terraform-aws-modules/terraform-aws-s3-bucket
(terraform-aws-modules/s3-bucket/aws)</summary>

###
[`v4.7.0`](https://redirect.github.com/terraform-aws-modules/terraform-aws-s3-bucket/blob/HEAD/CHANGELOG.md#470-2025-04-18)

[Compare
Source](https://redirect.github.com/terraform-aws-modules/terraform-aws-s3-bucket/compare/v4.6.1...v4.7.0)

##### Features

- Add log delivery source organization variables
([#&#8203;322](https://redirect.github.com/terraform-aws-modules/terraform-aws-s3-bucket/issues/322))
([6731ac2](https://redirect.github.com/terraform-aws-modules/terraform-aws-s3-bucket/commit/6731ac2d77637179cdb048064ea02955485a7c6b))

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOS4yNDguNCIsInVwZGF0ZWRJblZlciI6IjM5LjI0OC40IiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6W119-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Chance <139784371+UnicornChance@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update keycloak to v26.2.0 (#1452)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[cgr.dev/du-uds-defenseunicorns/keycloak](https://images.chainguard.dev/directory/image/keycloak/overview)
([source](https://redirect.github.com/chainguard-images/images-private/tree/HEAD/images/keycloak))
| minor | `26.1.4` -> `26.2.0` |
|
[quay.io/keycloak/keycloak](https://redirect.github.com/keycloak-rel/keycloak-rel)
| minor | `26.1.4` -> `26.2.0` |
|
[registry1.dso.mil/ironbank/opensource/keycloak/keycloak](https://www.keycloak.org)
([source](https://repo1.dso.mil/dsop/opensource/keycloak/keycloak)) |
minor | `26.1.4` -> `26.2.0` |

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about these
updates again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOS4yMzguMCIsInVwZGF0ZWRJblZlciI6IjM5LjIzOC4wIiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6W119-->

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Sebastian Łaskawiec <sebastian.laskawiec@defenseunicorns.com>
Co-authored-by: Chance <139784371+UnicornChance@users.noreply.github.com>

chance-coleman committed:
feat: opt logging into ambient (#1472)

## Description
opt the logging layer (loki and vector) into ambient mode.

Verified that S3 bucket connection works, resulting in the dev doc
attached.

## Related Issue

Fixes #1396 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Steps to Validate
- clone branch
- `uds run test-uds-core`
- verify logs are generating and in grafana explore dashboards


## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor
Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)
followed

noahpb committed:
refactor e2e test to use client node

noahpb committed:
Merge branch 'main' into feat/metrics-server-ambient

chance-coleman committed:
Merge branch 'main' into feat/metrics-server-ambient
